src/services/formatting/formatting.ts

```diff
@@ -97,11 +97,23 @@ namespace ts.formatting {
     }
 
     export function formatOnSemicolon(position: number, sourceFile: SourceFile, rulesProvider: RulesProvider, options: FormatCodeSettings): TextChange[] {
-        return formatOutermostParent(position, SyntaxKind.SemicolonToken, sourceFile, options, rulesProvider, FormattingRequestKind.FormatOnSemicolon);
+        const outermostParent = findOutermostParentWithinListLevelFromPosition(position, SyntaxKind.SemicolonToken, sourceFile);

```

Comment:
From out tests, consider
``` ts
if (true) {
if (true)
let x = 10
}
```
On typing a semi-colon after `10`, it's reasonable to format not only the statement, but the enclosing conditionals, yielding
``` ts
if (true) {
    if (true)
        let x = 10;
}
```